src/meta-srv/src/procedure/drop_table.rs

```diff
@@ -0,0 +1,271 @@
+use client::Database;
+use common_error::ext::ErrorExt;
+use common_error::status_code::StatusCode;
+use common_meta::ident::TableIdent;
+use common_meta::instruction::Instruction;
+use common_meta::rpc::ddl::DropTableTask;
+use common_meta::rpc::router::TableRoute;
+use common_meta::table_name::TableName;
+use common_procedure::error::{FromJsonSnafu, ToJsonSnafu};
+use common_procedure::{
+    Context as ProcedureContext, LockKey, Procedure, Result as ProcedureResult, Status,
+};
+use common_telemetry::debug;
+use futures::future::join_all;
+use serde::{Deserialize, Serialize};
+use snafu::{ensure, ResultExt};
+use table::engine::TableReference;
+use table::metadata::RawTableInfo;
+
+use super::utils::handle_retry_error;
+use crate::ddl::DdlContext;
+use crate::error;
+use crate::error::Result;
+use crate::procedure::utils::{build_table_metadata, handle_request_datanode_error, TableMetadata};
+use crate::service::mailbox::BroadcastChannel;
+use crate::service::store::txn::{Txn, TxnOp};
+use crate::table_routes::fetch_table;
+pub struct DropTableProcedure {
+    context: DdlContext,
+    data: DropTableData,
+}
+
+// TODO(weny): removes in following PRs.
+#[allow(unused)]
+impl DropTableProcedure {
+    pub(crate) const TYPE_NAME: &'static str = "metasrv-procedure::DropTable";
+
+    pub(crate) fn new(
+        cluster_id: u64,
+        task: DropTableTask,
+        table_route: TableRoute,
+        table_info: RawTableInfo,
+        context: DdlContext,
+    ) -> Self {
+        Self {
+            context,
+            data: DropTableData::new(cluster_id, task, table_route, table_info),
+        }
+    }
+
+    pub(crate) fn from_json(json: &str, context: DdlContext) -> ProcedureResult<Self> {
+        let data = serde_json::from_str(json).context(FromJsonSnafu)?;
+        Ok(Self { context, data })
+    }
+
+    /// Removes the table metadata.
+    async fn on_remove_metadata(&mut self) -> Result<Status> {
+        let table_ref = self.data.table_ref();
+
+        // If metadata not exists (might have already been removed).
+        if fetch_table(&self.context.kv_store, table_ref)
+            .await?
+            .is_none()
+        {
+            self.data.state = DropTableState::InvalidateTableCache;
+
+            return Ok(Status::executing(true));
+        }
+
+        let table_ref = self.data.table_ref();
+        let table_id = self.data.task.table_id;
+
+        let TableMetadata {
+            table_global_key,
+            table_global_value,
+            table_route_key,
+            table_route_value,
+        } = build_table_metadata(
+            table_ref,
+            self.data.table_route.clone(),
+            self.data.table_info.clone(),
+        )?;
+
+        let txn = Txn::new().and_then(vec![
+            TxnOp::Delete(table_route_key.to_string().into_bytes()),
+            TxnOp::Delete(table_global_key.to_string().into_bytes()),
+        ]);
+        let resp = self.context.kv_store.txn(txn).await?;
```

Comment:
I wouldn't recommend using transaction. If table is to be deleted, it's somehow not that important. Just simply use more lightweight `kv_store.delete`.